File: app/src/main/java/com/enipro/data/remote/model/ObjectId.java
package com.enipro.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.ParcelConstructor;

@org.parceler.Parcel
public class ObjectId {

    @SerializedName("$oid")
    @Expose
    public String $oid;

//    public ObjectId(Parcel in){
//        this.$oid = in.readString();
//    }

    @ParcelConstructor
    public ObjectId(String $oid){this.$oid = $oid;}

//    @Override
//    public void writeToParcel(Parcel dest, int flags) {
//        dest.writeString($oid);
//    }
//
//    @Override
//    public int describeContents() {
//        return 0;
//    }

    public String get_$oid() {
        return $oid;
    }

    public void set_$oid(String $oid) {
        this.$oid = $oid;
    }

//    public static final Parcelable.Creator<ObjectId> CREATOR = new Parcelable.Creator<ObjectId>() {
//        @Override
//        public ObjectId createFromParcel(Parcel source) {
//            return new ObjectId(source);
//        }
//
//        @Override
//        public ObjectId[] newArray(int size) {
//            return new ObjectId[size];
//        }
//    };
}

File: app/src/main/java/com/enipro/data/remote/model/ObjectId.kt
package com.enipro.data.remote.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ObjectId
constructor(@field:SerializedName("\$oid")
            @field:Expose
            var oid: String) : Parcelable {
    constructor(parcel: Parcel) : this(parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(oid)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ObjectId> {
        override fun createFromParcel(parcel: Parcel): ObjectId {
            return ObjectId(parcel)
        }

        override fun newArray(size: Int): Array<ObjectId?> {
            return arrayOfNulls(size)
        }
    }
}